bin-ranges-diff/src/main/java/uk/gov/pay/java_lambdas/bin_ranges_diff/App.java

```diff
@@ -1,17 +1,101 @@
 package uk.gov.pay.java_lambdas.bin_ranges_diff;
 
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.ResponseInputStream;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.utils.Pair;
+import uk.gov.pay.java_lambdas.bin_ranges_diff.util.RequestBuilder;
+import uk.gov.pay.java_lambdas.common.bin_ranges.dto.Candidate;
 
-public class App {
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static uk.gov.pay.java_lambdas.bin_ranges_diff.config.Constants.PROMOTED_BIN_RANGES_S3_KEY;
+import static uk.gov.pay.java_lambdas.bin_ranges_diff.config.Constants.S3_PROMOTED_BUCKET_NAME;
+import static uk.gov.pay.java_lambdas.bin_ranges_diff.config.Constants.S3_STAGING_BUCKET_NAME;
+
+public class App implements RequestHandler<Candidate, Candidate> {
     private static final Logger logger = LoggerFactory.getLogger(App.class);
+    private final S3Client s3Client;
+
+    public App() {
+        s3Client = DependencyFactory.s3Client();
+    }
+
+    @Override
+    public Candidate handleRequest(Candidate candidate, final Context context) {
+        logger.info("fn: {}, version: {}.", context.getFunctionName(), context.getFunctionVersion());
+        var digests = getShaDigestsFromS3(candidate.s3Key());
+        logger.info("Object SHAs [candidate: {}] [promoted: {}]", digests.left(), digests.right());
+        return digests.left().equals(digests.right()) 
+            ? Candidate.from(candidate, false)
+            : Candidate.from(candidate, true);
+    }
+
+    private Pair<String, String> getShaDigestsFromS3(String candidateKey) {
+
+//        var promotedBinRangesHeadObject = s3Client.headObject(RequestBuilder.headObjectRequest(S3_PROMOTED_BUCKET_NAME, PROMOTED_BIN_RANGES_S3_KEY));
+//        var existingDigest = Optional.ofNullable(promotedBinRangesHeadObject.metadata().get("sha256MessageDigest"));
+//        
+//        if (existingDigest.isPresent()) {
+//            
+//        }
+
+        var getCandidateBinRangesRequest = RequestBuilder.getObjectRequest(S3_STAGING_BUCKET_NAME, candidateKey);
+        var getPromotedBinRangesRequest = RequestBuilder.getObjectRequest(S3_PROMOTED_BUCKET_NAME, PROMOTED_BIN_RANGES_S3_KEY);
```

Comment:
Again, `var` is making me feel like I’m reading JavaScript!

Reply:
I'm of the opinion that `var` is acceptable in cases where the assignment is obvious and the variable is named appropriately. 

I don't think `GetObjectRequest getCandidateBinRangesRequest = RequestBuilder.getObjectRequest(...)` is any more descriptive here.

Reply:
At the time I wrote the comment, I was not aware there is a type called `GetObjectRequest`. Now I am so this makes more sense. Maybe the variables could be called `candidateBinRangesGetObjectRequest` and `promotedBinRangesGetObjectRequest` but that would be a bit on the long side.